Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MptrXmlElementAccess.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataformat.access;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import org.kitodo.api.dataformat.mets.LinkedMetsResource;
+import org.kitodo.dataformat.metskitodo.DivType;
+import org.kitodo.dataformat.metskitodo.DivType.Mptr;
+
+/**
+ * Provides access to the {@code <mets:mptr>} element.
+ */
+class MptrXmlElementAccess {
+    /**
+     * String indicating that the LOCTYPE value is other than one of the allowed
+     * constants and has been placed in the OTHERLOCTYPE element.
+     */
+    private static final String OTHER = "OTHER";
+
+    /**
+     * Set of allowed LOCTYPE values.
+     */
+    private static final Set<String> ALLOWED_LOCTYPE_VALUES = new HashSet<>(
+            Arrays.asList("ARK", "URN", "URL", "PURL", "HANDLE", "DOI", OTHER));
+
+    /**
+     * Adds the information about a linked METS resource to a
+     * {@code <mets:div>}.
+     *
+     * @param link
+     *            Details of a linked METS resource
+     * @param div
+     *            {@code <mets:div>} to which the information is added
+     */
+    static void addMptrToDiv(LinkedMetsResource link, DivType div) {
+        div.setORDER(link.getOrder());
+        Mptr mptr = new Mptr();
+        if (ALLOWED_LOCTYPE_VALUES.contains(link.getLoctype())) {
+            mptr.setLOCTYPE(link.getLoctype());
+        } else {
+            mptr.setLOCTYPE(OTHER);
+            mptr.setOTHERLOCTYPE(Objects.toString(link.getLoctype()));
+        }
+        mptr.setHref(link.getUri().toASCIIString());
+        div.getMptr().add(mptr);
+    }
+
+    /**
+     * Reads the information about a link from a {@code <mets:div>}.
+     *
+     * @param div
+     *            {@code <mets:div>}, which may contain information about a link
+     * @return information to the link, if indicated
+     */
+    static LinkedMetsResource readLink(DivType div) {
```

Comment:
Is
getLinkFromDiv
a better name?
Or do I get the method wrong? becuase you are not actually reading the link, right? You extract the information about the link from the div.

Reply:
Can one also say so.